jwst/pipeline/calwebb_spec2.py

```diff
@@ -277,29 +273,22 @@ def process_exposure_product(
             self.cube_build.output_type = 'multi'
             self.cube_build.suffix = 's3d'
             self.cube_build.save_results = False
-            cube = self.cube_build(input)
-            self.save_model(cube[0], 's3d')
-
-            # Pass the cube along for input to 1D extraction
-            x1d_input = cube.copy()
-            cube.close()
-
-        else:
-            # Pass the unresampled cal product to 1D extraction
-            x1d_input = input
+            rectified_result = self.cube_build(result)
+            self.save_model(rectified_result[0], 's3d')
 
         # Extract a 1D spectrum from the 2D/3D data
-        self.extract_1d.suffix = 'x1d'
         if tso_mode:
             self.extract_1d.suffix = 'x1dints'
-        x1d_output = self.extract_1d(x1d_input)
+        else:
+            self.extract_1d.suffix = 'x1d'
+        x1d_result = self.extract_1d(rectified_result)
```

Comment:
I don't see a path to ``extract_1d`` for data sets that don't get rectified. Right now the only exp_type's that produce a "rectified_result" are NIRSpec fixed-slit and MSA, MIRI LRS fixed-slit, MIRI MRS, and NIRSpec IFU. This leaves out exposure types like MIR_LRS-SLITLESS, NRC_WFSS, NIS_WFSS, NIS_SOSS, and NRS_BRIGHTOBJ, all of which don't get resampled, but do need to go through ``extract_1d`` (in unresampled form).